### PR TITLE
add NcML previewer #17

### LINKED ISSUE #17: NcML (XML) previewer
As part of the following issue and pull request...

- https://github.com/IQSS/dataverse/issues/9153
- https://github.com/IQSS/dataverse/pull/9239

... I'd like to add a previewer for NcML (XML) files that are derived from NetCDF and HDF5 files.

Right now it looks like this:

Screen Shot 2022-12-20 at 2 26 47 PM

### PR DESCRIPTION
**Pull request by pdurbin**
- Closes #17

Here's a screenshot of how it looks:

Screen Shot 2022-12-20 at 2 26 47 PM

### CODE REVIEW

**Review by qqmyers — CHANGES_REQUESTED** [review]
Didn't run this but it follows the basic pattern of all the other ones. I left a couple requests and a question about XSS filtering.

File: 5.2curlcommands.md [hunk 1]
```diff
@@ -579,3 +579,65 @@ curl -X POST -H 'Content-type: application/json' http://localhost:8080/api/admin
...
+
+```bash
+curl -X POST -H 'Content-type: application/json' http://localhost:8080/api/admin/externalTools -d \
+'{
+  "displayName":"Show NcML (XML)",
+  "description":"Metadata from HDF5 files.",
+  "toolName":"ncmlPreviewer",
+  "scope":"file",
+  "types":["preview"],
+  "toolUrl":"https://gdcc.github.io/dataverse-previewers/previewers/v1.3/NcmlPreview.html",
```
[the remark this answers is not in the document]
Comment (pdurbin) [reply 5.2curlcommands.md:591]: Sounds good. As discussed I [updated the wiki](https://github.com/gdcc/dataverse-previewers/wiki/How-to-create-a-previewer/_compare/266efc11c04182c039315836b633d3860d568dd9...3829a7cd26364b16511ca4d7c70c9f96be642349) and will tweak the README.

File: previewers/v1.3/js/ncml.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+$(document).ready(function() {
+    startPreview(false);
+});
+
+function translateBaseHtmlPage() {
+    var ncmlPreviewText = $.i18n( "ncmlPreviewText" );
+    $( '.ncmlPreviewText' ).text( ncmlPreviewText );
+}
+
+function writeContentAndData(data, fileUrl, file, title, authors) {
```
[the remark this answers is not in the document]
Comment (pdurbin) [reply previewers/v1.3/js/ncml.js]: Done in a0ee97c.

File: previewers/v1.3/js/ncml.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    addStandardPreviewHeader(file, title, authors);
+    $('.preview').append($("<textarea/>").html(data)
```
[the remark this answers is not in the document]
Comment (pdurbin) [reply previewers/v1.3/js/ncml.js]: Sure, I'm not exactly sure how it works but I copied it from text.js to ncml.js in 955ca61 and the XML still renders. Please take a look.

**Comment by pdurbin** [comment]
Because I can't put this PR on https://github.com/orgs/IQSS/projects/2 I'm using the following issue to represent its state:

- https://github.com/IQSS/dataverse/issues/9242

I'm about to move it back to review since I believe I've addressed the feedback so far.

While I'm here I'll leave a screenshot of the expected error page, which I was just testing:

Screen Shot 2022-12-22 at 11 46 13 AM

**Review by qqmyers — APPROVED** [review]

**Comment by qqmyers** [comment]
Looks good. The XSS filtering might be more than needed (i.e. the html previewer filters much less than the text previewer), but that can be adjusted as people play with real NetCDF files and decide what's useful.

**Pull request merged**